iOS-13-Migration-Guide.md

```diff
@@ -45,7 +45,76 @@ If your App supports incoming calls, you **MUST** perform the following steps to
+4. You must register via `[TwilioVoice registerWithAccessToken:deviceToken:completion:]` when your App starts.
+
+	**Swift**
+	
+	```.swift
+	// AppDelegate.swift
+	class AppDelegate: UIResponder, UIApplicationDelegate, PKPushRegistryDelegate {
+	    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
+	        ...
+	        self.setupPushRegistry()
+	        ...
+	    }
+
+	    func setupPushRegistry() {
+	        var voipRegistry: PKPushRegistry
+	        voipRegistry.delegate = self
+            voipRegistry.desiredPushTypes = Set([PKPushType.voIP])
+	    }
+
+	    func pushRegistry(_ registry: PKPushRegistry, didUpdate credentials: PKPushCredentials, for type: PKPushType) {
+	        let accessToken = fetchAccessToken()
+            let deviceToken = credentials.token.map { String(format: "%02x", $0) }.joined()
+
+            TwilioVoice.register(withAccessToken: accessToken, deviceToken: deviceToken) { (error) in
+                ...
+            }
+	    }
+	}
+	```
+
+	**Objective-C**
+
+	```.objective-c
+	// AppDelegate.m
+	@interface AppDelegate () <PKPushRegistryDelegate>
+	@property (nonatomic, strong) PKPushRegistry *voipRegistry;
+	@end
+
+	@implementation AppDelegate
+
+	- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+	    ...
+	    [self setupPushRegistry];
+	    ...
+	}
+
+	- (void)setupPushRegistry {
+	    self.voipRegistry = [[PKPushRegistry alloc] initWithQueue:dispatch_get_main_queue()];
+	    self.voipRegistry.delegate = self;
+	    self.voipRegistry.desiredPushTypes = [NSSet setWithObject:PKPushTypeVoIP];
+	}
+
+	#pragma mark - PKPushRegistryDelegate
+	- (void)pushRegistry:(PKPushRegistry *)registry didUpdatePushCredentials:(PKPushCredentials *)credentials forType:(NSString *)type {
+	    self.deviceTokenString = [credentials.token description];
+	    NSString *accessToken = [self fetchAccessToken];
+
+	    [TwilioVoice registerWithAccessToken:accessToken
+	                             deviceToken:self.deviceTokenString
+	                              completion:^(NSError *error) {
+	        ...
+	    }];
+	}
+
+	@end
+	```
+
+This ensures that your app no longer receives “cancel” push notifications. Please note that if the app is updated but never launched for the execution of the above, the mobile client will still receive "cancel" notifications, which could cause the app terminated by iOS if the VoIP push notification is not reported to CallKit as a new incoming call.
```

Comment:
I would create a section under the code snippet that describes what to do to manage the corner case where the app has not been launched yet. Additionally in this section I would actually provide a code snippet with the recommended workaround.